filecoin-proofs/src/api/mod.rs

```diff
@@ -318,6 +323,153 @@ where
     add_piece(source, target, piece_size, Default::default())
 }
 
+// Verifies if a DiskStore specified by a config is consistent.
+fn verify_store(config: &StoreConfig, arity: usize) -> bool {
+    let store_path = StoreConfig::data_path(&config.path, &config.id);
+    if !Path::new(&store_path).exists() {
+        return false;
+    }
+
+    if config.size.is_none() {
+        return false;
+    }
+
+    let res = DiskStore::<<DefaultPieceHasher as Hasher>::Domain>::is_consistent(
+        config.size.unwrap(),
+        arity,
+        &config,
+    )
+    .is_ok();
+    trace!("disk store {:?} is_consistent: {}", store_path, res);
+
+    res
+}
+
+// Checks for the existence of the tree d store, the replica, and all generated labels.
+pub fn validate_cache_for_precommit_phase2<R, T>(
+    cache_path: R,
+    replica_path: T,
+    seal_precommit_phase1_output: &SealPreCommitPhase1Output,
+) -> Result<bool>
+where
+    R: AsRef<Path>,
+    T: AsRef<Path>,
+{
+    let replica = replica_path.as_ref().to_path_buf();
+    if !Path::new(&replica).exists() {
+        trace!("replica {:?} does not exist", &replica);
+        return Ok(false);
+    }
+
+    for label in &seal_precommit_phase1_output.labels.labels {
```

Comment:
I think it would a bit better to have this be a method no the `Labels` struct, so we don't forget to update the logic here, when that changes